aptos-move/framework/aptos-framework/sources/simple_delegations.move

```diff
@@ -0,0 +1,844 @@
+        );
+
+        // Initialize the stake pool in a new resource account. This allows the same delegator to delegate to multiple
+        // different operators.
+        let seed = create_seed(delegator_address, operator);
+        let (stake_pool_signer, _) = account::create_resource_account(delegator, seed);
+        // Initialize stake pool with amount = 0 as the coins need to come from the delegator, not the stake pool
+        // resource account.
+        stake::initialize_stake_owner(&stake_pool_signer, 0, operator, voter);
+
+        // Extract owner_cap from the StakePool, so we have control over it in the delegations flow.
+        // This is stored as part of the delegation. Thus, the delegator would not have direct control over it without
+        // going through well-defined functions in this module.
+        let owner_cap = stake::extract_owner_cap(&stake_pool_signer);
+
+        // Add the delegation to the stake pool.
+        let delegated_coins = coin::withdraw<AptosCoin>(delegator, amount);
+        stake::add_stake_with_cap(&owner_cap, delegated_coins);
+
+        // Record the delegation.
+        let pool_address = signer::address_of(&stake_pool_signer);
+        simple_map::add(delegations_map, operator, Delegation {
+            is_active: true,
+            total_stake: amount,
+            pool_address,
+            owner_cap,
+            commission_percentage,
+            debts: vector::empty<CommissionDebt>(),
+        });
+
+        emit_event<DelegateEvent>(
+            &mut delegations.delegate_events,
+            DelegateEvent { operator, voter, pool_address, amount, commission_percentage },
+        );
+    }
+
+    /// Convenient function to allow the delegator to update the voter address in a delegation they made.
+    public entry fun update_voter(delegator: &signer, operator: address, new_voter: address) acquires Delegations {
+        let delegator_address = signer::address_of(delegator);
+        assert_delegation_exists(delegator_address, operator);
+
+        let delegations = borrow_global_mut<Delegations>(delegator_address);
+        let delegations_map = &mut delegations.delegations;
+        let delegation = simple_map::borrow_mut(delegations_map, &operator);
+        let pool_address = delegation.pool_address;
+        let old_voter = stake::get_delegated_voter(pool_address);
+        stake::set_delegated_voter_with_cap(&delegation.owner_cap, new_voter);
+
+        emit_event<UpdateVoterEvent>(
+            &mut delegations.update_voter_events,
+            UpdateVoterEvent { operator, pool_address, old_voter, new_voter },
+        );
+    }
+
+    /// Unlocks commission amount from the stake pool. Operator needs to wait for the amount to become withdrawable
+    /// at the end of the stake pool's lockup period before they can actually can withdraw_commission.
+    ///
+    /// Anyone can call this function as this is an explicit agreement between delegator and operator.
```

Comment:
it feels very weird that anyone can call this function, intuitively this should require the operator signer since it can prevent auto-compounding for example

Reply:
It's weird but my rationale is that this allows the previous operator to call and trigger paying their debts as well. But I missed the compounding part which is a really good point - Anyone can decide how much compounding will be done with the commissions, which seems scary to the operator. I updated both request_commission and withdraw_commission to allow only the operator to call.